Wikipedia/UI-V5/WMFArticleContainerViewController.m

```diff
@@ -433,13 +508,21 @@ - (void)saveWebViewScrollOffset {
 - (void)fetchArticle {
     @weakify(self);
     [self unobserveArticleUpdates];
-    self.articleFetcherPromise = [self.articleFetcher fetchArticleForPageTitle:self.articleTitle progress:NULL]
-                                 .then(^(MWKArticle* article) {
+    [self showProgressViewAnimated:YES];
+    self.articleFetcherPromise = [self.articleFetcher fetchArticleForPageTitle:self.articleTitle progress:^(CGFloat progress) {
+        dispatchOnMainQueue(^{
+            [self updateProgress:[self totalProgressWithArticleFetcherProgress:progress] animated:YES];
+        });
+    }].then(^(MWKArticle* article) {
         @strongify(self);
         [self saveWebViewScrollOffset];
+        dispatchOnMainQueue(^{
```

Comment:
yeah - they are needed to get the UI updating progressively - for some reason just doing git in line tends to delay getting updates until the very end.